### PR TITLE
More VFS and logging updates

### PR DESCRIPTION
**Pull request by h0tw1r3**
- refactor all load/save lists with utils helpers
- fix amt scraper watched to update the correct list depending if streaming or downloading
- fix moviequiz integration (current release does not accept type as int)

*\* Tested on 12.2 release.

### CODE REVIEW

**Comment by h0tw1r3** [comment]
I have lots of plans for the script.  For example, currently working on all new "quiz" implementation.  If you're too busy to commit/comment (or have no interest), please let me know and I'll fork under a new name instead.

**Comment by Giftie** [comment]
Hi..  So that I have missed this, my Github notifications get lost with all the XBMC ones.  

The biggest problems I have with this is that you have pulled two other branches into the master, filesystem and logging_update and too many changes for a single pull request.  

Each of the branches should first be worked on separately, ie logging update first, then filesystem.  This makes less chances of things breaking.  Also your master should always match my master of the script, not having it this way can cause many conflict errors and headaches(my GitHub-fu is not that great :) ).. I have brought both branches inline with the master.  If you can, and I know this creates more work, make the changes to those branches and submit pull requests for those changes.  I know I stopped working on the filesystem branch as the xbmc.File() stuff was not 100%.  

Your changes to the Apple trailer scraper is very much appreciated... I dread web page scraping :)  I think it is no longer necessary to use script.module.elementtree  as is it is included in Python 2.6(which is the lowest Python XBMC uses) -> http://forum.xbmc.org/showthread.php?tid=173887&pid=1510396#pid1510396

I really do appreciate the enthusiasm, don't get me wrong... The help is really appreciated.   

Any additions to the script that help make it extraordinary are always welcome.  

I hope I don't scare you off..  Right now I am concentrating on a settings revamp with a GUI version instead of the XBMC integrated Addon Settings.  It will allow a lot more flexibility and customization( and hopefully less confusion too)

**Comment by h0tw1r3** [comment]
Didn't realize elementtree was included.  Thanks!

Not scared off, but it is a bit confusing because there are commits that look like cherry-picks, and branch commits not related to its feature.  For example, quiz was broken in my environment because I make heavy use of path substitution, but it was fixed only by merging both logging_update and filesystem.

What branching model are you using?  From what I can tell, merging master into the feature branch is the opposite of what you have been doing & it's really hard to follow now.  Were some things rebased?  The sha's look different now.

`git log --graph --pretty=format:'%h %d %s %cr <%an>' --all` looks weird.

I will try to finish the log branch and filesystem branch and submit pull requests.

**Comment by Giftie** [comment]
The weirdness is probably caused by my Git-Fu...  I had some conflicts between master and the branches.  Primarily I rebase the branch to master, fetch master as upstream and fast forward all the branch's commits. I think the conflicts were caused by the fact I fixed some of the errors twice, once in the branch then forgetting that it was fixed, fixed it slightly different in the master.
